pkg/northbound/gnmi/v2/get.go

```diff
@@ -344,3 +349,47 @@ func (s *Server) checkOpaAllowed(ctx context.Context, targetInfo *targetInfo, co
 	}
 	return modelPlugin.GetPathValues(ctx, "", []byte(bodyText))
 }
+
+func (s *Server) reportAllTargets(ctx context.Context, encoding gnmi.Encoding) (*gnmi.GetResponse, error) {
+	// Get list of configurable entities from the topo store
+	targetEntities, err := s.topo.List(ctx, &topoapi.Filters{
+		ObjectTypes: []topoapi.Object_Type{topoapi.Object_ENTITY},
+		WithAspects: []string{"onos.topo.Configurable"},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Distill the list of configurable entities into their corresponding target IDs
+	targets := make([]string, 0, len(targetEntities))
+	for _, targetEntity := range targetEntities {
+		targets = append(targets, string(targetEntity.ID))
+	}
+
+	// Produce an appropriately encoded path value with all target IDs
+	var allDevicesPathElem = make([]*gnmi.PathElem, 0)
+	allDevicesPathElem = append(allDevicesPathElem, &gnmi.PathElem{Name: "all-devices"})
```

Comment:
Should we use target terminology instead of device terminology ?

Reply:
Yes, we should! Good catch! :)